src/Reporter.cc

```diff
@@ -537,10 +537,12 @@ void Reporter::DoLog(const char* prefix, EventHandlerPtr event, FILE* out, Conne
 
         // Enlarge buffer;
         size *= 2;
-        buffer = allocated = (char*)realloc(allocated, size);
+        buffer = (char*)realloc(allocated, size);
 
         if ( ! buffer )
             FatalError("out of memory in Reporter");
+        else
+            allocated = buffer;
```

Comment:
No need for the else here.

Reply:
Wow, looking around, I guess it might make sense to try and switch from the following four variables to a `std::string` approach and assume whatever optimization is being done here isn't needed :-)

```
    static char tmp[512];
    int size = sizeof(tmp);
    char* buffer = tmp;
    char* allocated = nullptr;

```

Reply:
Looking at this again, you actually don't need to repeatedly make this call either. Passing a null buffer to vsnprintf will return the size of the buffer needed, which you can then allocate one time instead of realloc'ing repeatedly.

Reply:
I made a change to only call `realloc()` once if needed. We can look at the performance differences of using std::string here separately.